drivers/pci/pci.c

```diff
@@ -406,4 +408,25 @@ void BootPciPeripheralInitialization()
 	// frankenregisters so Xromwell matches Cromwell
 	PciWriteDword(BUS_1, DEV_0, FUNC_0, 0x0c, 0x0);
 	PciWriteDword(BUS_1, DEV_0, FUNC_0, 0x18, 0x08);
+
+	// Select serial device
+	IoOutputByte(0x2e, 0x07);
+	IoOutputByte(0x2f, 0x04);
+
+	// Enable device
+	IoOutputByte(0x2e, 0x30);
+	IoOutputByte(0x2f, 0x01);
+
+	// Set Serial Base
+	IoOutputByte(0x2e, 0x61);
+	IoOutputByte(0x2f, SERIAL_PORT & 0xff);
+
+	IoOutputByte(0x2e, 0x60);
+	IoOutputByte(0x2f, SERIAL_PORT >> 8);
+
+	// Set Serial Interrupt
+	IoOutputByte(0x2e, 0x70);
+	IoOutputByte(0x2f, SERIAL_IRQ);
+
+	IoOutputByte(0x2e, 0xAA);	// Exit Configuration
```

Comment:
There's a lot of redundancy with menu/actions/PhyMenuActions.c; abstract into separate file?

Reply:
Refactored.